packages/mui-joy/src/styles/CssVarsProvider.tsx

```diff
@@ -7,20 +7,40 @@ import { unstable_createCssVarsProvider as createCssVarsProvider } from '@mui/sy
 import defaultTheme from './defaultTheme';
 import type { SupportedColorScheme } from './types';
 import THEME_ID from './identifier';
+import { defaultConfig } from '../InitColorSchemeScript/InitColorSchemeScript';
 
-const { CssVarsProvider, useColorScheme, getInitColorSchemeScript } = createCssVarsProvider<
-  SupportedColorScheme,
-  typeof THEME_ID
->({
+const {
+  CssVarsProvider,
+  useColorScheme,
+  getInitColorSchemeScript: deprecatedGetInitColorSchemeScript,
+} = createCssVarsProvider<SupportedColorScheme, typeof THEME_ID>({
   themeId: THEME_ID,
   theme: defaultTheme,
-  attribute: 'data-joy-color-scheme',
-  modeStorageKey: 'joy-mode',
-  colorSchemeStorageKey: 'joy-color-scheme',
+  attribute: defaultConfig.attribute,
+  modeStorageKey: defaultConfig.modeStorageKey,
+  colorSchemeStorageKey: defaultConfig.colorSchemeStorageKey,
   defaultColorScheme: {
-    light: 'light',
-    dark: 'dark',
+    light: defaultConfig.defaultLightColorScheme,
+    dark: defaultConfig.defaultDarkColorScheme,
   },
 });
 
+let warnedInitScriptOnce = false;
+
+const getInitColorSchemeScript: typeof deprecatedGetInitColorSchemeScript = (params) => {
```

Comment:
@siriwatknp 
How about adding `@deprecated` jsdoc here?
A warning in IDE is a good developer experience, IMO.
What do you think?